packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -198,6 +199,138 @@ class RenderEditable extends RenderBox {
 
   Rect _lastCaretRect;
 
+  bool _rightArrowDown = false;
+  bool _leftArrowDown = false;
+  bool _upArrowDown = false;
+  bool _downArrowDown = false;
+
+  static const int _kLeftArrowCode = 21;
+  static const int _kRightArrowCode = 22;
+  static const int _kUpArrowCode = 19;
+  static const int _kDownArrowCode = 20;
+
+  int _extentOffset = -1;
+  int _baseOffset = -1;
+
+  int _previousCursorLocation = -1;
+  bool _resetCursor = false;
+
+  void _keyEvent(RawKeyEvent keyEvent){
+    if (!(keyEvent.data is RawKeyEventDataAndroid))
+      return;
+
+    final RawKeyEventDataAndroid rawAndroidEvent = keyEvent.data;
+    final int pressedKeyCode = rawAndroidEvent.keyCode;
+
+    if (selection.isCollapsed) {
+      _extentOffset = selection.extentOffset;
+      _baseOffset = selection.baseOffset;
+    }
+
+    // Update current key states
+    final bool pressedDown = keyEvent is RawKeyDownEvent;
```

Comment:
Do we actually need to do anything for keyup? If so, is the logic for keyDown and keyUp disjoint enough that we could separate it into different methods?

Reply:
I mean, all key up does is change the value of the arrow down booleans. Because the conditional will only execute on key down.